nncf/api/compression.py

```diff
@@ -150,6 +151,70 @@ def __add__(self, other: 'CompressionStage') -> 'CompressionStage':
         return CompressionStage.PARTIALLY_COMPRESSED
 
 
+class CompressionSetup:
+    """
+    Consists of builder and controller states - a dictionaries with Python data structures,
+    defining how to setup the compression
+    """
+    BUILDER_STATE_ATTR = 'builder_state'
+    CONTROLLER_STATE_ATTR = 'ctrl_state'
+
+    def __init__(self, builder_state: Dict, ctrl_state: Dict):
+        self.builder_state = builder_state
+        self.ctrl_state = ctrl_state
+
+    def get_state(self) -> Dict:
+        """
+        :return: a dictionary with Python data structures (dict, list, tuple, str, int, float, True, False, None) that
+        represents state of the object.
+        """
+        return {
+            self.BUILDER_STATE_ATTR: self.builder_state,
+            self.CONTROLLER_STATE_ATTR: self.ctrl_state
+        }
+
+    @classmethod
+    def from_state(cls, state: Dict) -> 'CompressionSetup':
+        """
+        Creates the object from its state.
+        :param state: Output of `get_state()` method.
+        """
+        return cls(**state)
+
+
+class CompressionState:
+    """
```

Comment:
I was looking for the word that can describe combination of controller and builder state.
I chose `CompressionSetup` because builder and controller in a way setup compression. 
`Composite` and `Algorithm` are not suitable here. Call it `ControllerState` or `BuilderState` also isn't right.

Maybe simple call it `CtrlAndBuilderStates`? 😄

Reply:
A controller and a builder together form an algorithm, why isn't `Algorithm` or `Algo` suitable?

Reply:
technically, builder and controller states are not the whole algorithm state. algorithm produces extra compression parameters stored in a model state

Reply:
maybe 
builder state + controller state = compression state
compression state + model state = nncf checkpoint?

Reply:
kept `CompressionState` only